Add request options to work with newer OTP

dbernheisel opened the pull request:
When using newer OTP versions, there are more options we may need to specify when making HTTPS requests, such as limiting the cipher suite to only using TLS 1.2.

ryanflach commented:
@dbernheisel since all of the tests are utilizing stubs for the calls, what's the best way to test this locally?

dbernheisel commented:
Sorry, yeah it'd be helpful to have some ideas on how to test! I'll also update the PR description to be more.. descriptive.

Here's how I verified:

On master using a newer version of Elixir like 1.11.3-otp-23, run this:
```sh
iex -S mix
```

```elixir
iex> ExVatcheck.check("ATU28560205")
** (Protocol.UndefinedError) protocol String.Chars not implemented for {:options, {:sslv3, {:versions, [:"tlsv1.2", :"tlsv1.1", :tlsv1, :sslv3]}}} of type Tuple. This protocol is implemented for the following type(s): Integer, Version.Requirement, Atom, List, DateTime, Float, BitString, Time, URI, NaiveDateTime, Date, Version
    (elixir 1.11.3) lib/string/chars.ex:3: String.Chars.impl_for!/1
    (elixir 1.11.3) lib/string/chars.ex:22: String.Chars.to_string/1
    (ex_vatcheck 0.1.5) lib/ex_vatcheck/vies_client.ex:49: ExVatcheck.VIESClient.new/0
    (ex_vatcheck 0.1.5) lib/ex_vatcheck.ex:29: ExVatcheck.validate/1
```

then if you update HTTPoison/hackney:

```sh
mix deps.update httpoison
```

```elixir
iex> ExVatcheck.check("ATU28560205")
%ExVatcheck.VAT{
  exists: true,
  valid: true,
  vies_available: true,
  vies_response: %{
    address: "Tobelbader Straße 30\nAT-8141 Premstätten",
    country_code: "AT",
    name: "ams AG",
    request_date: "2021-03-19",
    valid: true,
    vat_number: "U28560205"
  }
}
```

and try to run it again it'll work. So really the only thing we need to do is update the mix.lock with a newer hackney. This branch does that.

---

Another change is limiting the ciphers to only trust tls1.2 and tls1.3 since those are still worthwhile, anything earlier is effectively broken. The VIES site supports tls1.2 and tls1.3 ([ssllabs report](https://www.ssllabs.com/ssltest/analyze.html?d=ec.europa.eu&s=147.67.210.30&hideResults=on&ignoreMismatch=on))

---

Since you bring this up, it kinda shows that our tests for this lib is not that great. CI should be able to at least smoke-test a single request, and do that across versions of OTP to ensure the connection is good. I'll see if I can get that in.

dbernheisel commented:
Added these changes to help w/ testing:

- Traded Travis CI for GitHub Actions since travis-ci.org is going away.
- There's a new smoke test that makes a real request to verify the HTTP connection. This test is excluded by default, but CI includes it.

Doing this revealed some issues that I fixed now:
- Updated TLS1.3 to only be included on OTP 23. The rest will use TLS1.2.
- Increased the required Elixir version. CI failed on Elixir 1.4.x due to some library requirements. I decided since we can't test it at that level, might as well drop support for it.

Because of these changes, I updated the changelog to be 0.2.0 instad of 0.1.6 since that's probably a breaking change for someone.

dbernheisel commented on lib/ex_vatcheck/vies_client.ex:
```diff
@@ -7,6 +7,11 @@ defmodule ExVatcheck.VIESClient do
   alias ExVatcheck.VIESClient.XMLParser
 
   @wsdl_url "https://ec.europa.eu/taxation_customs/vies/checkVatService.wsdl"
+  if System.otp_release() == "23" do
```
this will make OTP 24 not use tls1.3 which doesn't seem right. I'll update this to use a `Version.compare`

poops reviewed (approved):
Code LGTM! I was able to test the failure locally by using this branch's `.tool-versions` in master, and then confirmed the failing VAT check works in this branch.
